logic.js

```diff
@@ -0,0 +1,221 @@
+        
+        
+        var img = document.createElement("img");
+        img.src = data[id]["image"];
+        img.alt = "pizza_img:" + data[id]["id"];
+        img.classList.add("img_fit");
+
+        pizzaImg.appendChild(img);
+        pizza.appendChild(pizzaImg);
+
+
+        var pizzaInfo =  document.createElement("div");
+        pizzaInfo.classList.add("pizza_info");
+        
+        var pizza_title = document.createElement("div");
+        pizza_title.classList.add("pizza_title");
+        pizza_title.innerHTML = data[id]["title"];
+        pizzaInfo.appendChild(pizza_title);
+
+        var pizza_ingredients = document.createElement("div");
+
+        var pizza_ingredients_array  = data[id]["ingredients"];
+
+
+        var ing="";
+        pizza_ingredients_array.forEach(element => {
+
+            if(ing === ""){
+                ing =element;
+            }
+            else{
+             ing =ing + ", " +element;
+            }
+        });
+        ing=ing.replace("/\n/"," ");
+
+        ing="Składniki:  " + ing;
+
+        pizza_ingredients.innerHTML=  ing;
+        pizza_ingredients.classList.add("pizza_ingredients");
+        pizzaInfo.appendChild(pizza_ingredients);
+
+        var pizza_price = document.createElement("div");
+        pizza_price.classList.add("pizza_price");
+        pizza_price.innerHTML="Cena: " + Number(data[id]["price"]).toFixed(2) + " zł";
+        pizzaInfo.appendChild(pizza_price);
+
+        pizza.appendChild(pizzaInfo);
+
+        var buttonAdd = document.createElement("button");
+        buttonAdd.innerHTML = "zamów";
+        buttonAdd.addEventListener("click", AddToChart);
+        pizza.appendChild(buttonAdd);
+    
+        return pizza;
+    }
+
+
+    function AddToChart(event){
+
+        if(CalculateAmount()>49){
+            alert("W przypadku zamówień powyżej 50 sztuk prosimy o kontakt telefoniczny");
+        
+        }
+        else{
+
+            var currentID = event.target.parentNode.id;
+
+            if (chart.find(e => e.id == currentID)) {
+                chart.find(e => e.id == currentID).amount +=1;
+            }
+            else{
+                chart.push({id: data[currentID-1]["id"], amount:1 });
+            }
+
+            RefreshChart();
+        }
+    }
+
+    function RemoveFromChart(event){
+        var currentID = event.target.pizzaID;
+
+        var current = chart.find(e => e.id == currentID);
+
+        if(current.amount>1){
+            current.amount-=1;
+        }
+        else{
+           var index = chart.indexOf(current);
+            if (index > -1) {
+                chart.splice(index, 1);
+            }
+
+        }
+        RefreshChart();
+
+    }
+
+    function RefreshChart(){
+
+        price_sum_count = 0;
+
+
+        while (chartSpace.childNodes.length>2) {
+            chartSpace.removeChild(chartSpace.lastChild);
+        }
+    
+        if(chart.length > 0){
+            empty_chart_text.style.display="none";
```

Comment:
nie zmieniaj styli bezposrednio w js, dodaj/skasuje klase ktora jest odpowiedzialna za to

Reply:
poprawione